### PR TITLE
chore: review safety comments

### PR DESCRIPTION
**Pull request by DaniPopes**

### CODE REVIEW

File: crates/revm/src/inspector/instruction.rs [hunk 1]
```diff
@@ -1,49 +1,39 @@
...
-            // step
-            let data = &mut host.data;
-            // Safety: as the PC was already incremented we need to subtract 1 to preserve the
-            // old Inspector behavior.
-            unsafe {
-                interpreter.instruction_pointer = interpreter.instruction_pointer.sub(1);
-            }
             if let Some(inspector) = host.inspector.as_mut() {
-                inspector.step(interpreter, data);
+                // Safety: as the PC was already incremented we need to subtract 1 to preserve the
```
Comment (lorenzofero) [inline crates/revm/src/inspector/instruction.rs:16]:
```suggestion
                // SAFETY: as the PC was already incremented we need to subtract 1 to preserve the
```
According to https://std-dev-guide.rust-lang.org/policy/safety-comments.html#inside-safe-elements

File: crates/interpreter/src/instructions/system.rs [hunk 2]
```diff
@@ -91,7 +90,6 @@ pub fn calldatacopy<H: Host>(interpreter: &mut Interpreter<'_>, _host: &mut H) {
     let data_offset = as_usize_saturated!(data_offset);
     shared_memory_resize!(interpreter, memory_offset, len);
 
-    // Safety: set_data is unsafe function and memory_resize ensures us that it is safe to call it
```
Comment (rakita) [inline crates/interpreter/src/instructions/system.rs:94]: Can you revert this?

File: crates/interpreter/src/instructions/system.rs [hunk 3]
```diff
@@ -45,7 +45,6 @@ pub fn codecopy<H: Host>(interpreter: &mut Interpreter<'_>, _host: &mut H) {
     let code_offset = as_usize_saturated!(code_offset);
     shared_memory_resize!(interpreter, memory_offset, len);
 
-    // Safety: set_data is unsafe function and memory_resize ensures us that it is safe to call it
```
Comment (rakita) [inline crates/interpreter/src/instructions/system.rs:48]: Can you revert this?

File: crates/interpreter/src/interpreter/stack.rs [hunk 4]
```diff
@@ -40,7 +41,7 @@ impl Stack {
     #[inline]
     pub fn new() -> Self {
         Self {
-            // Safety: [`Self::push`] assumes that capacity is STACK_LIMIT
+            // Important: expansion functions assume that capacity is `STACK_LIMIT`.
```
Comment (rakita) [inline crates/interpreter/src/interpreter/stack.rs:44]: Would still put `SAFETY` here

File: crates/revm/src/journaled_state.rs [hunk 5]
```diff
@@ -183,7 +183,7 @@ impl JournaledState {
     /// 5. Increment nonce of created account if SpuriousDragon is active
     /// 6. Decrease balance of caller account.
     ///
-    /// Safety: It is assumed that caller balance is already checked and that
+    /// Note: it is assumed that caller balance is already checked and that
```
Comment (rakita) [inline crates/revm/src/journaled_state.rs:186]: Would rephrase this as a `# Panic`

File: crates/interpreter/src/instructions/system.rs (hunk 2 above)
Comment (DaniPopes) [reply crates/interpreter/src/instructions/system.rs:94]: the function is not unsafe, but I'll keep the comment as a note
